sled-agent/src/artifact_store.rs

```diff
@@ -601,30 +590,27 @@ impl ArtifactWriter {
             }
             // 4. fsync the parent directory.
             if let Err(err) = parent_dir.sync_all().await {
-                log_and_store!(
-                    self.last_error,
-                    &self.log,
-                    "sync",
-                    mountpoint,
-                    err
-                );
+                log_and_store!(last_error, &self.log, "sync", mountpoint, err);
                 continue;
             }
 
-            any_success = true;
+            successful_writes += 1;
```

Comment:
There's an expectation here that "we're writing a single file" to each M.2, right?

I think that's okay, just clarifying, because, I think our expected output is:

- datasets = 2, successful_writes = 2

And it would be a little confusing to suddenly see:

- datasets = 2, successful_writes = 4

or something like that, if we started writing / syncing multiple files